lib/util.js

```diff
@@ -400,11 +400,14 @@ function formatValue(ctx, value, recurseTimes) {
     });
   }
 
+  var constructor = getConstructorOf(value);
+
   // Some type of object without properties can be shortcutted.
   if (keys.length === 0) {
     if (typeof value === 'function') {
-      return ctx.stylize(`[Function${value.name ? `: ${value.name}` : ''}]`,
-                         'special');
+      const cName = constructor ? constructor.name : 'Function';
```

Comment:
I have an irrational dislike for the name `cName`.  `constructorName` or `ctorName` if the former is too long?

(Applies to line 541 as well.)